test/test_binary_ufuncs.py

```diff
@@ -1359,72 +1359,79 @@ def test_rdiv(self, device, dtype):
+                    fn(x, zero)
+            # Different value for different dtype on CUDA:
+            # Due to it's an undefined behavior, CUDA returns a pattern of all 1s
+            # for integral dividend (other than int64) divided by zero. For int64,
+            # CUDA returns all 1s for negative dividend, half 1s for positive dividend.
+            # uint8: 0xff -> 255
+            # int32: 0xffffffff -> -1
             else:
-                value = 255 if dtype == torch.uint8 else -1
-                self.assertTrue(torch.all(x.fmod(zero) == value))
+                if dtype == torch.int64:
+                    self.assertEqual(fn(x, zero) == 4294967295, x >= 0)
+                    self.assertEqual(fn(x, zero) == -1, x < 0)
+                else:
+                    value = 255 if dtype == torch.uint8 else -1
+                    self.assertTrue(torch.all(fn(x, zero) == value))
 
     @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False, include_bool=False, include_complex=False))
-    def test_fmod(self, device, dtype):
+    def test_fmod_remainder(self, device, dtype):
         # Use numpy as reference
         def _helper(x, mod):
-            np_x = x.cpu().numpy()
-            np_mod = mod.cpu().numpy() if torch.is_tensor(mod) else mod
-            exp = np.fmod(np_x, np_mod)
-            exp = torch.from_numpy(exp)
-
-            res = torch.fmod(x, mod)
-            # Different Type promotion between numpy and torch for
-            # integral tensor with float scalar/tensor
-            dtype_flag = \
-                dtype in torch.testing.get_all_int_dtypes() and \
-                (mod.dtype == torch.float if torch.is_tensor(mod) else type(mod) == float)
-
-            if dtype_flag:
-                self.assertEqual(res, exp, exact_dtype=False)
-            else:
-                self.assertEqual(res, exp)
-            # out
-            out = torch.empty(0, device=device, dtype=dtype)
-            if dtype_flag:
-                self.assertRaises(RuntimeError, lambda: torch.fmod(x, mod, out=out))
-            else:
-                torch.fmod(x, mod, out=out)
-                self.assertEqual(out, exp)
-                self.assertEqual(out.size(), torch.Size([10, 10]))
-            # in-place
-            if dtype_flag:
-                self.assertRaises(RuntimeError, lambda: x.fmod_(mod))
-            else:
-                x.fmod_(mod)
-                self.assertEqual(x, exp, exact_dtype=False)
+            fns_list = ((torch.fmod, torch.Tensor.fmod_, np.fmod),
+                        (torch.remainder, torch.Tensor.remainder_, np.remainder))
+            for fn, inplace_fn, ref in fns_list:
+                np_x = x.cpu().numpy()
+                np_mod = mod.cpu().numpy() if torch.is_tensor(mod) else mod
+                exp = ref(np_x, np_mod)
+                exp = torch.from_numpy(exp)
+
+                res = fn(x, mod)
+                # Different Type promotion between numpy and torch for
```

Comment:
To enforce consistent integer->float type promotion you can wrap the NumPy references with

https://github.com/pytorch/pytorch/blob/573f4aa3529371e9d30a0884d56eac0e6d3ddd71/torch/testing/_internal/common_methods_invocations.py#L284

Then you wouldn't have to deal with this